medusa/tv/series.py

```diff
@@ -2077,6 +2077,98 @@ def __qualities_to_string(qualities=None):
         return ', '.join([Quality.qualityStrings[quality] for quality in qualities or []
                           if quality and quality in Quality.qualityStrings]) or 'None'
 
+    def want_season(self, season, quality, forced_search=False, download_current_quality=False):
+        """Check if we we want the episodes included withing the multi-ep search result."""
+
+        wanted_episodes = []
+        # if the quality isn't one we want under any circumstances then just say no
+        allowed_qualities, preferred_qualities = self.current_qualities
+        log.debug(
+            u'{id}: Allowed, Preferred = [ {allowed} ] [ {preferred} ] Found = [ {found} ]', {
+                'id': self.indexerid,
+                'allowed': self.__qualities_to_string(allowed_qualities),
+                'preferred': self.__qualities_to_string(preferred_qualities),
+                'found': self.__qualities_to_string([quality]),
+            }
+        )
+
+        if not Quality.wanted_quality(quality, allowed_qualities, preferred_qualities):
+            log.debug(
+                u"{id}: Ignoring found result for '{show}' S{season} with unwanted quality '{quality}'", {
+                    'id': self.indexerid,
+                    'show': self.name,
+                    'season': season,
+                    'quality': Quality.qualityStrings[quality],
+                }
+            )
+            return []
+
+        main_db_con = db.DBConnection()
+        sql_results = main_db_con.select(
+            b'SELECT '
+            b'  episode, '
+            b'  status, '
+            b'  manually_searched '
+            b'FROM '
+            b'  tv_episodes '
+            b'WHERE '
+            b'  showid = ? '
+            b'  AND season = ? ', [self.indexerid, season])
+
+        if not sql_results or not len(sql_results):
+            log.debug(
+                u'{id}: Unable to find a episodes matching the season searched in database.'
+                u' Ignoring found result for {show} S{season} with quality {quality}', {
+                    'id': self.indexerid,
+                    'show': self.name,
+                    'season': season,
+                    'quality': Quality.qualityStrings[quality],
+                }
+            )
+            return []
+
+        for episode in sql_results:
+            # should_replace = None
```

Comment:
@p0psicles as i already said before. you are looping in ALL episodes of the season just to check if quality is desired when in a season pack all episodes have the same quality, so you only need to check one.

@duramato